TTS/tts/layers/delightful_tts/acoustic_model.py

```diff
@@ -0,0 +1,591 @@
+        speaker_embedding = None
+        emotion_embedding = None
+        if d_vectors is not None:
+            speaker_embedding = torch.nn.functional.normalize(d_vectors)
+
+        if emo_vectors is not None:
+            emotion_embedding = torch.nn.functional.normalize(emo_vectors)
+
+        # Encoder
+        pos_encoding = positional_encoding(
+            self.emb_dim,
+            max(token_embeddings.shape[1], max(mel_lens)),
+            device=token_embeddings.device,
+        )
+        encoder_outputs = self.encoder(
+            token_embeddings,
+            src_mask,
+            speaker_embedding=speaker_embedding,
+            emotion_embedding=emotion_embedding,
+            encoding=pos_encoding,
+        )
+
+        # Utterance Prosody encoder
+        u_prosody_ref = self.u_norm(self.utterance_prosody_encoder(mels=mels, mel_lens=mel_lens))
+        u_prosody_pred = self.u_norm(
+            self.average_utterance_prosody(
+                u_prosody_pred=self.utterance_prosody_predictor(x=encoder_outputs, mask=src_mask),
+                src_mask=src_mask,
+            )
+        )
+
+        if use_ground_truth:
+            encoder_outputs = encoder_outputs + self.u_bottle_out(u_prosody_ref)
+        else:
+            encoder_outputs = encoder_outputs + self.u_bottle_out(u_prosody_pred)
+
+        # Phoneme prosody encoder
+        p_prosody_ref = self.p_norm(
+            self.phoneme_prosody_encoder(
+                x=encoder_outputs, src_mask=src_mask, mels=mels, mel_lens=mel_lens, encoding=pos_encoding
+            )
+        )
+        p_prosody_pred = self.p_norm(self.phoneme_prosody_predictor(x=encoder_outputs, mask=src_mask))
+
+        if use_ground_truth:
+            encoder_outputs = encoder_outputs + self.p_bottle_out(p_prosody_ref)
+        else:
+            encoder_outputs = encoder_outputs + self.p_bottle_out(p_prosody_pred)
+
+        encoder_outputs_res = encoder_outputs
+
+        # Pitch predictor
+        pitch_pred, avg_pitch_target, pitch_emb = self.pitch_adaptor.get_pitch_embedding_train(
+            x=encoder_outputs,
+            target=pitches,
+            dr=dr,
+            mask=src_mask,
+        )
+
+        # Energy predictor
+        energy_pred, avg_energy_target, energy_emb = self.energy_adaptor.get_energy_embedding_train(
+            x=encoder_outputs,
+            target=energies,
+            dr=dr,
+            mask=src_mask,
+        )
+
+        encoder_outputs = (
+            encoder_outputs.transpose(1, 2) + pitch_emb + energy_emb
+        )  # [B, T_src, C_hidden] --> [B, C_hidden, T_src]
+
+        # Duration predictor
+        # log_duration_prediction = self.duration_predictor(txt_enc=encoder_outputs_res.detach(), lens=src_lens)
+        log_duration_prediction = self.duration_predictor(x=encoder_outputs_res.detach(), mask=src_mask)
+        # duration_ranges = self.range_predictor(x=encoder_outputs_res, dr=dr, x_lengths=src_lens)
```

Comment:
We don't use the range predictor and gaussian upsampling despite the implementations. Let's just remove those in the commented segments or uncomment them and set a toggle in the config to activate them if needed.

Reply:
okay yea, theres a lot of commented out code I need to clean up next